crates/y-sweet-worker/src/server_context.rs

```diff
@@ -16,8 +16,18 @@ pub struct ServerContext {
 impl ServerContext {
     pub fn new(config: Configuration, env: &Env) -> Self {
         let bucket = env.bucket(&config.bucket).unwrap();
-        let store = R2Store::new(bucket, config.bucket_prefix.clone());
-        let store: Arc<Box<dyn Store>> = Arc::new(Box::new(store));
+        let store: Box<dyn Store> = if let Some(ref s3) = config.s3_store_config {
```

Comment:
I think `if let Some(s3) = config.s3_store_config.as_ref()` should be preferred here since it tells the compiler that you are not transferring ownership of `config.s3_store_config`. It doesn't matter in this case because it's never used again, but it's more consistent with how we want the code to be read.